presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -284,6 +283,7 @@ private void schedule()
                         .collect(toImmutableList()));
                 if (queryStateMachine.isDone()) {
                     sectionExecutions.forEach(SectionExecution::abort);
+                    break;
```

Comment:
I assume if we clean up `executionSchedules` list here, we should also be able to break the loop via the condition form 276-278? :)

Reply:
no the problem is that next time through the loop we'll consider this section "ready for execution" again because the most recent attempt is in a failed state and this might be a retry.

Reply:
i see